src/plopp/backends/matplotlib/image.py

```diff
@@ -1,251 +1,33 @@
-from ...graphics.bbox import BoundingBox, axis_bounds
-from ...graphics.colormapper import ColorMapper
-from ..common import check_ndim
-from .canvas import Canvas
-
-
-def _find_dim_of_2d_coord(coords):
-    for xy, coord in coords.items():
-        if coord['var'].ndim == 2:
-            return (xy, coord['dim'])
-
-
-def _get_dims_of_1d_and_2d_coords(coords):
-    dim_2d = _find_dim_of_2d_coord(coords)
-    if dim_2d is None:
-        return None, None
-    axis_1d = 'xy'.replace(dim_2d[0], '')
-    dim_1d = (axis_1d, coords[axis_1d]['dim'])
-    return dim_1d, dim_2d
-
-
-def _maybe_repeat_values(data, dim_1d, dim_2d):
-    if dim_2d is None:
-        return data
-    return repeat(data, dim=dim_1d[1], n=2)[dim_1d[1], :-1]
-
+from .fast_image import FastImage
+from .mesh_image import MeshImage
 
-def _from_data_array_to_pcolormesh(data, coords, dim_1d, dim_2d):
-    z = _maybe_repeat_values(data=data, dim_1d=dim_1d, dim_2d=dim_2d)
-    if dim_2d is None:
-        return coords['x'], coords['y'], z
 
-    # Broadcast 1d coord to 2d and repeat along 1d dim
-    # TODO: It may be more efficient here to first repeat and then broadcast, but
-    # the current order is simpler in implementation.
-    broadcasted_coord = repeat(
-        sc.broadcast(
-            coords[dim_1d[0]],
-            sizes={**coords[dim_2d[0]].sizes, **coords[dim_1d[0]].sizes},
-        ).transpose(data.dims),
-        dim=dim_1d[1],
-        n=2,
-    )
-    # Repeat 2d coord along 1d dim
-    repeated_coord = repeat(coords[dim_2d[0]].transpose(data.dims), dim=dim_1d[1], n=2)
-    out = {dim_1d[0]: broadcasted_coord[dim_1d[1], 1:-1], dim_2d[0]: repeated_coord}
-    return out['x'], out['y'], z
-
-
-class Image:
+def Image(
+    *,
+    data: sc.DataArray,
+    **kwargs,
+):
     """
-    Artist to represent two-dimensional data.
+    Factory function to create an image artist.
+    If all the coordinates of the data are 1D and linearly spaced,
+    a `FastImage` is created.
+    Otherwise, a `MeshImage` is created.
 
     Parameters
     ----------
-    canvas:
-        The canvas that will display the image.
-    colormapper:
-        The colormapper to use for the image.
     data:
-        The initial data to create the image from.
-    uid:
-        The unique identifier of the artist. If None, a random UUID is generated.
-    shading:
-        The shading to use for the ``pcolormesh``.
-    rasterized:
-        Rasterize the mesh/image if ``True``.
-    **kwargs:
-        Additional arguments are forwarded to Matplotlib's ``pcolormesh``.
+        The data to create the image from.
     """
-
-    def __init__(
-        self,
-        canvas: Canvas,
-        colormapper: ColorMapper,
-        data: sc.DataArray,
-        uid: str | None = None,
-        shading: str = 'auto',
-        rasterized: bool = True,
-        **kwargs,
+    if all(
+        (data.coords[dim].ndim < 2)
+        and ((data.coords[dim].dtype == str) or (sc.islinspace(data.coords[dim])))
```

Comment:
This requires that there are dim-coords for each dim. (Because it uses `data.coords[dim]`.) Was this a requirement already before?

Reply:
Yes. If the input is missing them, some dummy (arange) coords are added before you get to here.
There are tests for this. See e.g. https://github.com/scipp/plopp/blob/69a60a3a12c699a0489f6f4a084eb07aba7b96ce/tests/plotting/plot_1d_test.py#L51